e2e/tests/components/number-picker/readonly/number-picker-read-only.e2e-spec.ts

```diff
@@ -0,0 +1,93 @@
+import { browser } from 'protractor';
+import { Key } from 'selenium-webdriver';
+import { imageCompare } from '../../common/image-compare';
+import { NumberPickerUpdateOnPage } from './number-picker-read-only.po.spec';
+
+describe('Number Picker Tests readonly:', () => {
+
+    const page: NumberPickerUpdateOnPage = new NumberPickerUpdateOnPage();
+
+    beforeAll(async () => {
+        await page.getPage();
+    });
+
+    describe('true', () => {
+
+        it('should prevent changes to the value via the input', async () => {
+            await browser.actions().mouseMove(page.numberPickerReadOnly).click().perform();
+            await browser.actions().sendKeys('7').perform();
+
+            expect(await page.emittedValue.getText()).toBe('{ "readonly": 5 }');
+            expect(await imageCompare('number-picker-readonly-initial')).toEqual(0);
+        });
+
+        it('should prevent changes to the value by hitting the increment arrow', async () => {
+            await browser.actions().mouseMove(page.incrementArrow).click().perform();
+
+            expect(await page.emittedValue.getText()).toBe('{ "readonly": 5 }');
+        });
+
+        it('should prevent changes to the value by hitting the decrement arrow', async () => {
+            await browser.actions().mouseMove(page.decrementArrow).click().perform();
+
+            expect(await page.emittedValue.getText()).toBe('{ "readonly": 5 }');
+        });
+
+        it('should prevent changes to the value by hitting the up arrow on the keyboard', async () => {
+            await browser.actions().mouseMove(page.numberPickerReadOnly).click().perform();
+            await browser.actions().sendKeys(Key.ARROW_UP).perform();
+
+            expect(await page.emittedValue.getText()).toBe('{ "readonly": 5 }');
+        });
+
+        it('should prevent changes to the value by hitting the down arrow on the keyboard', async () => {
+            await browser.actions().mouseMove(page.numberPickerReadOnly).click().perform();
+            await browser.actions().sendKeys(Key.ARROW_DOWN).perform();
+
+            expect(await page.emittedValue.getText()).toBe('{ "readonly": 5 }');
+        });
```

Comment:
The missing test case here is the scroll wheel.

Reply:
It appears that protractor has a hard time simulating scroll events, it doesn't have any built in support. I tried using `browser.executeScript('window.scrollBy(0,100);');` on the element but it doesn't seem to change the value when readonly is set to false. 
Is there any e2e tests that I could use for help? I had a look but couldn't find anything.

Reply:
I believe you will have to do it using a karma test, there is a specific event for the mousewheel which is not invoked just by scrolling.
https://developer.mozilla.org/en-US/docs/Web/API/Element/wheel_event

Reply:
Awesome thanks!